frontend/src/services/normService.ts

```diff
@@ -150,22 +153,84 @@ export function usePutNormXml(
  * Returns the norm from the API. Reloads when the parameters change.
  *
  * @param eli ELI of the norm
- * @param options Optional additional filters and queries. This is only a
- *  placeholder for now and can not have a value.
+ * @param options Optional additional filters and queries
  * @param [fetchOptions={}] Optional configuration for fetch behavior
  * @returns Reactive fetch wrapper
  */
 export function useNormService(
   eli: MaybeRefOrGetter<string>,
-  options?: never,
+  options?: {
+    /**
+     * Render metadata in the HTML preview. Note that this is only applicable
+     * if you get the HTML preview, and will fail on other requests.
+     */
+    showMetadata?: boolean
+    /**
+     * Render the HTML preview at a specific date. Note that this is only
+     * applicable if you get the HTML preview, and will fail on other requests.
+     */
+    at?: MaybeRefOrGetter<Date | undefined>
+  },
   fetchOptions: UseFetchOptions = {},
 ): UseFetchReturn<Norm> {
   const url = computed(() => {
     const eliVal = toValue(eli)
     if (!eliVal) return INVALID_URL
 
-    return `/norms/${eliVal}`
+    const queryParams = new URLSearchParams()
+
+    if (options?.showMetadata) {
+      queryParams.append("showMetadata", "true")
+    }
+
+    if (options?.at) {
+      const atVal = toValue(options?.at)
+      if (atVal instanceof Date) {
+        queryParams.append("atIsoDate", atVal.toISOString())
+      }
+    }
+
+    return `/norms/${eliVal}?${queryParams.toString()}`
   })
 
-  return useApiFetch<Norm>(url, fetchOptions).json()
+  return useApiFetch<Norm>(url, fetchOptions)
+}
+
+/**
+ * Convenience shorthand for `useNormService` that sets the correct
+ * configuration for getting JSON data.
+ *
+ * @param eli ELI of the norm
+ * @param options Optional additional filters and queries
+ * @param [fetchOptions={}] Optional configuration for fetch behavior
+ * @returns Reactive fetch wrapper
+ */
+export const useGetNorm: typeof useNormService = (
+  eli,
+  options,
+  fetchOptions,
+) => {
+  return useNormService(eli, options, fetchOptions).json()
+}
+
+/**
+ * Convenience shorthand for `useNormService` that sets the correct
+ * configuration for getting the HTML preview.
+ *
+ * @param eli ELI of the norm
+ * @param options Optional additional filters and queries
+ * @param [fetchOptions={}] Optional configuration for fetch behavior
+ * @returns Reactive fetch wrapper
+ */
+export function useGetNormHtml(
+  eli: Parameters<typeof useNormService>["0"],
+  options?: Parameters<typeof useNormService>["1"],
+  fetchOptions?: Parameters<typeof useNormService>["2"],
+): UseFetchReturn<string> {
+  return useNormService(eli, options, {
+    ...fetchOptions,
+    beforeFetch(c) {
+      c.options.headers = { ...c.options.headers, Accept: "text/html" }
```

Comment:
if we have the option to pass the options we can also directly set the header in there. But I'm a bit unsure if thats a better option.

Reply:
Not sure what you mean, can you give me an example?

Reply:
```ts
return useNormService(eli, {
  ...options,
  headers: {
    ...options.headers,
    Accept: "text/html"
  }
}, fetchOptions)
```